lib/Basic/Subprocess.cpp

```diff
@@ -1056,11 +1056,16 @@ void llbuild::basic::spawnProcess(
     assert(readfds[0].fd == outputPipeParentEnd.unsafeDescriptor());
     assert(readfds[1].fd == controlPipeParentEnd.unsafeDescriptor());
 
-    if (poll(readfds, nfds, -1) == -1) {
-      int err = errno;
-      delegate.processHadError(ctx, handle,
-          Twine("failed to poll (") + strerror(err) + ")");
-      break;
+    while (poll(readfds, nfds, -1) == -1) {
+        int err = errno;
+
+        if (err == EAGAIN || err == EINTR) {
+          continue;
+        } else {
+          delegate.processHadError(ctx, handle,
+            Twine("failed to poll (") + strerror(err) + ")"); 
+          goto pollError;
```

Comment:
Can we just return here?   Why the `goto` ?

Reply:
I guess we can. I just wanted to have the exact same behaviour as before where we returned to after the while loop

Reply:
I think the more direct return is better in this case and matches the coding style more cleanly.